include/lala/logic/ternarize.hpp

```diff
@@ -272,10 +315,13 @@ class Ternarizer
         || ((f.sig() == MIN || f.sig() == MAX) && is_boolean(t1) && is_boolean(t2)))
       {
         t0 = toplevel ? ternarize_constant(F::make_z(1)) : introduce_bool_var();
-      }
-      else {
+      } 
+      else if (is_int(t1) && is_int(t2)) {
```

Comment:
probably safer to use `if(!is_real(t1) && !is_real(t2))` otherwise what if t1 or t2 is Boolean?